### PR TITLE
More gcc fixes, and get tests running

### PR DESCRIPTION
**Pull request by gonzoleeman**
This pull request fixes yet more gcc-10 issues, this time on s390 and ppc-64-bit. Also, after running the self test, I found an issue and fixed it.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,3 +3,5 @@ autom4te.cache
 config.*
 stamp-h1
 libtool
+tags
+cscope.files
```
Comment (kamalesh-babulal) [inline .gitignore]: There might be other cscope generated files like `cscope.out`, can we make it `cscope.*`

File: test/test.c [hunk 2]
```diff
@@ -165,7 +165,7 @@ static int path_is_dir(const char *path)
 {
 	struct stat astats;
 
-	if ((lstat(path, &astats)) != 0)
+	if ((stat(path, &astats)) != 0)
```
Comment (kamalesh-babulal) [inline test/test.c:168]: Good catch. `path_is_file()` too needs to be fixed, to use `stat` instead of `lstat`

**Review by kamalesh-babulal — APPROVED** [review]

**Review by kamalesh-babulal — APPROVED** [review]

File: .gitignore (hunk 1 above)
Comment (gonzoleeman) [reply .gitignore]: I have pushed a commit that changes it to "cscope.*"

File: test/test.c (hunk 2 above)
Comment (gonzoleeman) [reply test/test.c:168]: I have pushed a commit that fixes path_is_file()

**Comment by gonzoleeman** [comment]
Updated my branch, as per your comments.

**Pull request merged**